### PR TITLE
Include account name in Azure settings exceptions

### PR DESCRIPTION
**Pull request by DaveCTurner**
If there are several Azure accounts defined, and one of them has invalid
settings, then it's hard to pinpoint the problem from the messages given
today. This commit includes the account name in the exception message to
make troubleshooting easier.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

**Review by pxsalehi — APPROVED** [review]

File: modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java [hunk 1]
```diff
@@ -124,10 +124,6 @@ final class AzureStorageSettings {
     );
 
     private final String account;
-
-    @Nullable
-    private final String sasToken;
```
Comment (pxsalehi) [inline modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java:129]: actually one question. why is this removed?

Comment (pxsalehi) [reply modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java:129]: ok, I see that it was unused.
